probe/docker/container_test.go

```diff
@@ -56,11 +59,10 @@ func TestContainer(t *testing.T) {
 	if err = json.NewEncoder(writer).Encode(&stats); err != nil {
 		t.Error(err)
 	}
-	runtime.Gosched() // wait for StartGatheringStats goroutine to receive the stats
 
 	// Now see if we go them
-	nmd := c.GetNodeMetadata()
-	if nmd[docker.MemoryUsage] != "12345" {
-		t.Errorf("want 12345, got %s", nmd[docker.MemoryUsage])
-	}
+	test.Poll(t, 10*time.Millisecond, func() bool {
```

Comment:
What if there's a logical failure? (i.e. always != 12345). Then, the test will take 10s no matter what. I guess we don't want to optimize for failures but this can make things unpleasant when doing big refactorings.